src/3d/symbols/qgspointcloud3dsymbol.h

```diff
@@ -0,0 +1,102 @@
+/***************************************************************************
+  qgspointcloud3dsymbol.h
+  ------------------------------
+  Date                 : November 2020
+  Copyright            : (C) 2020 by Nedjima Belgacem
+  Email                : belgacem dot nedjima at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSPOINTCLOUD3DSYMBOL_H
+#define QGSPOINTCLOUD3DSYMBOL_H
+
+#include "qgis_3d.h"
+
+#include "qgsabstract3dsymbol.h"
+
+#define SIP_NO_FILE
+
+/**
+ * \ingroup 3d
+ * 3D symbol that draws point cloud geometries as 3D objects.
+ *
+ * \warning This is not considered stable API, and may change in future QGIS releases. It is
+ * exposed to the Python bindings as a tech preview only.
+ *
+ * \since QGIS 3.18
+ */
+class _3D_EXPORT QgsPointCloud3DSymbol : public QgsAbstract3DSymbol
+{
+  public:
+    //! Constructor for QgsPointCloud3DSymbol
+    QgsPointCloud3DSymbol() : QgsAbstract3DSymbol() {  }
+    ~QgsPointCloud3DSymbol() override = default;
+
+    QString type() const override { return "pointcloud"; }
+    QgsAbstract3DSymbol *clone() const override SIP_FACTORY
+    {
+      // TODO: use unique_ptr
+      QgsPointCloud3DSymbol *result = new QgsPointCloud3DSymbol;
+      result->mEnabled = mEnabled;
+      result->mPointSize = mPointSize;
+      copyBaseSettings( result );
+      return result;
+    }
+
+
+    void writeXml( QDomElement &elem, const QgsReadWriteContext &context ) const override
+    {
+      Q_UNUSED( context )
+
+      elem.setAttribute( QStringLiteral( "enabled" ), mEnabled );
+      elem.setAttribute( QStringLiteral( "point-size" ), mPointSize );
+    }
+
+    void readXml( const QDomElement &elem, const QgsReadWriteContext &context ) override
+    {
+      Q_UNUSED( context )
+
+      mEnabled = elem.attribute( "enabled", QStringLiteral( "0" ) ).toInt();
+      mPointSize = elem.attribute( "point-size", QStringLiteral( "5.0" ) ).toFloat();
+    }
+
+    /**
+     * Returns whether rendering for this symbol is enabled
+     * \see setIsEnabled( bool enabled )
+     * \since QGIS 3.18
```

Comment:
We usually don't include redundant \since tags if the member has been introduced in the same version as the class itself (it's effectively redundant because the class itself has the same \since tag)